### PR TITLE
Feature/students & Data Manager Consolidate

### PR DESCRIPTION
**Pull request by winner23**
The Pull request consists of three part:
1. Common DataManager for all entities 
2. The Entry-level structure for each entity (require testing and correction according to requirements)
3. Management of entity Student

### CODE REVIEW

File: DTAdmin.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -1,568 +1,688 @@
 // !$*UTF8*$!
 {
-    archiveVersion = 1;
```
Comment (iostaf) [inline DTAdmin.xcodeproj/project.pbxproj]: I'm just curious, what have you done to produce such indentation mess in `project.pbxproj` file?

File: DTAdmin/MainTabBarController.swift [hunk 2]
```diff
@@ -16,10 +16,11 @@ class MainTabBarController: UITabBarController, UITabBarControllerDelegate {
         var tabBarViewControllers = [UIViewController]()
         
 //MARK: Create students tab
-        let studentsTab = TabOneViewController()
-        let studentsBarItem = UITabBarItem(title: "Students", image: nil, selectedImage: nil)
-        studentsTab.tabBarItem = studentsBarItem
-        tabBarViewControllers.append(studentsTab)
+        let studentStoryboard = UIStoryboard(name: "Student", bundle: nil)
+        guard let studentNavigationController = studentStoryboard.instantiateViewController(withIdentifier: "StudentNavigationController") as? UINavigationController else { return }
+        let studentBarItem = UITabBarItem(title: "Students", image: nil, selectedImage: nil)
```
Comment (iostaf) [inline DTAdmin/MainTabBarController.swift]: Please use NSLocalizedString for all user faced messages.

File: DTAdmin/Models/DataManager.swift [hunk 3]
```diff
@@ -0,0 +1,312 @@
+//
+//  DataManager.swift
+//  DTAdmin
+//
+//  Created by Володимир on 10/24/17.
```
Comment (iostaf) [inline DTAdmin/Models/DataManager.swift]: Please do not use Cyrillic characters.

File: DTAdmin/Models/DataManager.swift [hunk 4]
```diff
@@ -0,0 +1,312 @@
...
+//  Copyright © 2017 if-ios-077. All rights reserved.
+//
+
+import Foundation
+
+class DataManager: HTTPManager {
+
+    static var shared = DataManager()
```
Comment (iostaf) [inline DTAdmin/Models/DataManager.swift]: Why not let instead of var?

File: DTAdmin/Student/Controllers/EditStudentViewController.swift [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+                    newStudent.userId = newUserId
+                    if let resultModification = self.resultModification {
+                        resultModification(newStudent, true)
+                    }
+                    self.navigationController?.popViewController(animated: true)
+                }
+            }
+        }
+    }
+    func prepareForSave() -> Bool {
```
Comment (iostaf) [inline DTAdmin/Student/Controllers/EditStudentViewController.swift]: It's not clear what does this function do? When it returns true and when false?
Please write Swift doc for it

File: DTAdmin/Student/Controllers/EditStudentViewController.swift [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+                self.emailStudentTextField.text = userInstance.email
+            } else if let error = error {
+                self.showWarningMsg(error)
+            }
+        }
+    }
+}
+extension UIImage
+{
+    func convert(toSize size:CGSize, scale:CGFloat) -> UIImage
```
Comment (iostaf) [inline DTAdmin/Student/Controllers/EditStudentViewController.swift]: Why do we need this function? What does it do?

File: DTAdmin/Student/Controllers/GroupsTableViewController.swift [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+    @IBOutlet var groupTable: UITableView!
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        DataManager.shared.getList(byEntity: .Group) { (groups, error) in
+            if error == nil,
+                let groups = groups as? [GroupStructure] {
+                self.groupList = groups
+                self.groupTable.reloadData()
+            } else {
+                self.showWarningMsg(error ?? "Incorect type data")
```
Comment (iostaf) [inline DTAdmin/Student/Controllers/GroupsTableViewController.swift]: Please use NSLocalizedString for all user faced messages.

File: DTAdmin/Student/Controllers/StudentViewController.swift [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+    @objc func updateTable(){
+        startActivity()
+        DataManager.shared.getList(byEntity: .Student) { (students, error) in
+            self.stopActivity()
+            if error == nil,
+                let students = students as? [StudentStructure] {
+                self.studentList = students
+                self.studentTable.reloadData()
+            } else {
+                self.showWarningMsg(error ?? "Incorect type data")
```
Comment (iostaf) [inline DTAdmin/Student/Controllers/StudentViewController.swift]: Please use NSLocalizedString for all user faced messages.

**Review by iostaf — CHANGES_REQUESTED** [review]

File: DTAdmin.xcodeproj/project.pbxproj (hunk 1 above)
Comment (winner23) [reply DTAdmin.xcodeproj/project.pbxproj]: I'm not sure that I made it manually...

**Review by iostaf — APPROVED** [review]

**Pull request merged**